### PR TITLE
Upgrade from OSS 2.x to Cloud

### PR DESCRIPTION
**Pull request by sanderson**
Related to #2332

Adds the upgrade guide for OSS 2.x to Cloud.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+{{% note %}}
+If you've created other tokens in your OSS 2.x instance for external libraries or
+integrations, create corresponding tokens for each in your InfluxDB Cloud instance.
+You cannot migrate tokens from InfluxDB OSS to InfluxDB Cloud.
+{{% /note %}}
+
+_For more information about managing tokens and token types, see [Manage tokens](/influxdb/cloud/security/tokens/)._
+
+## Set up influx CLI connection configurations
+The `influx` command line interface (CLI) lets you configure connection configurations
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:67]: create connection config...or configure connections?

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+that automatically provides **host**, **organization**, and **authentication token**
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:68]:
```suggestion
that automatically provide **host**, **organization**, and **authentication token**
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+credentials to CLI commands.
+Use the `influx` CLI packaged with InfluxDB OSS 2.x and the
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:70]:
```suggestion
Use the `influx` CLI packaged with InfluxDB 2.x and the
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+```sh
+# Example cloud connection configuration
+influx config create \
+  --config-name cloud \
+  --host-url https://cloud2.influxdata.com \
+  --org your.email@example.com \
+  --token mY5uP3rS3cRe7Cl0uDt0K3n
+```
+
+##### Create an InfluxDB OSS 2.x connection configuration
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:99]: We can prob drop OSS everywhere, otherwise, spell out in first mention

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+your **InfluxDB Cloud** instance.
+
+{{% note %}}
+#### InfluxDB Cloud Free Plan resource limits
+If upgrading to an [InfluxDB Cloud Free Plan](/influxdb/cloud/account-management/pricing-plans/#free-plan),
+you are only able to create a limited number of resources.
+If your exported template exceeds these limits, the resource migration will fail.
+{{% /note %}}
+
+- **To migrate _all resources_ from an InfluxDB OSS organization to an InfluxDB Cloud organization**:
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:124]: InfluxDB 2.0?

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+    {{< /code-tabs-wrapper >}}
+
+    For more export command examples, see the [`influx export`](/influxdb/cloud/reference/cli/influx/export/#examples)
+    and [`influx export all`](/influxdb/cloud/reference/cli/influx/export/all#examples) documentation.
+
+    {{% note %}}
+#### Update hardcoded InfluxDB URLs
+If any of your migrated resources contain hardcoded InfluxDB URLs (`http://localhost:8086`),
+update these URLs to your [InfluxDB Cloud region URL](/influxdb/cloud/reference/regions/).
+Do either of the following:
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:173]:
```suggestion
If any of your migrated resources contain hardcoded InfluxDB URLs (`http://localhost:8086`), do one of the following to update these URLs to your [InfluxDB Cloud region URL](/influxdb/cloud/reference/regions/):
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+
+- Update URLs in the InfluxDB Cloud UI after you've migrated your resources to your InfluxDB Cloud instance.
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:175]:
```suggestion
- Migrate your resources to InfluxDB Cloud, and then update URLs in the InfluxDB Cloud UI.
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+- Output your template to a file and then manually update the URLs in the file
+  before applying the template to your InfluxDB Cloud instance.
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:177]:
```suggestion
- Save your template to a file, update URLs in the file, and then apply the template to your InfluxDB Cloud instance.
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+    {{% /note %}}
+
+## Migrate DBRP mappings
+InfluxDB database and retention policy (DBRP) mappings let you query InfluxDB Cloud
+buckets with InfluxQL and the InfluxDB 1.x DBRP convention.
+**If you have DBRP mappings in your InfluxDB OSS 2.x instance**, manually migrate them
+to your InfluxDB Cloud instance.
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:184]:
```suggestion
**If you have DBRP mappings in your InfluxDB OSS 2.x instance**, migrate them
to your InfluxDB Cloud instance.
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 10]
```diff
@@ -0,0 +1,373 @@
...
+
+{{< expand-wrapper >}}
+{{% expand "Manually migrate DBRP mappings to InfluxDB Cloud"%}}
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:187]:
```suggestion
{{% expand "Migrate DBRP mappings to InfluxDB Cloud"%}}
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 11]
```diff
@@ -0,0 +1,373 @@
...
+      --active-config cloud \
+      --bucket-id 12ab34cd56ef \
+      --database example-db \
+      --rp example-rp
+    ```
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## Dual write to InfluxDB OSS 2.x and InfluxDB Cloud
+With buckets and DBRP mappings (if applicable) in place, you can update external
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:217]:
```suggestion
Update external
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 12]
```diff
@@ -0,0 +1,373 @@
...
+    - `INFLUX_TOKEN`: InfluxDB OSS 2.x authentication token
+    - `INFLUX_CLOUD_TOKEN`: InfluxDB Cloud authentication token
+
+3.  Use the command provided in your [Telegraf Setup Instructions](/influxdb/cloud/telegraf-configs/#use-influxdb-telegraf-configurations)
+    to restart Telegraf with the updated configuration and begin writing to both
+    InfluxDB OSS 2.x and InfluxDB Cloud.
+
+## Migrate time series data
+To migrate your time series data from your InfluxDB OSS 2.x instance to your
+InfluxDB Cloud instance:
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:261]:
```suggestion
To migrate your time series data from your InfluxDB OSS 2.x instance to your
InfluxDB Cloud instance, do the following:
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 13]
```diff
@@ -0,0 +1,373 @@
...
+    - **-\-file**: Import file path
+
+    ```sh
+    influx write \
+      --active-config cloud \
+      --bucket example-bucket \
+      --file path/to/bucket-export.lp
+    ```
+
+Repeat this process for each bucket.
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:305]:
```suggestion
4. Repeat steps 2-3 for each bucket.
```

File: content/influxdb/cloud/upgrade/from-v2.md [hunk 14]
```diff
@@ -0,0 +1,373 @@
...
+However, when writing to InfluxDB Cloud, **Data In** and **Ingest batch size**
+rate limits track the payload size of the **uncompressed** line protocol.
+{{% /note %}}
+
+#### Migrate system buckets
+InfluxDB [system buckets](/influxdb/cloud/reference/internals/system-buckets/)
+contain data related to the InfluxDB monitoring and alerting system.
+Although the retention period for system buckets in both InfluxDB Cloud and
+InfluxDB OSS 2.x is only seven days, if you want to migrate this data,
+use the same method described [above](#migrate-time-series-data).
```
Comment (kelseiv) [inline content/influxdb/cloud/upgrade/from-v2.md:347]:
```suggestion
use the same method described above [to migrate time series data](#migrate-time-series-data).
```

File: content/influxdb/v2.0/reference/glossary.md [hunk 15]
```diff
@@ -1055,8 +1055,8 @@ Related entries: [point](#point), [unix timestamp](#unix-timestamp), [RFC3339 ti
 Tokens (or authentication tokens) verify user and organization permissions in InfluxDB.
 There are different types of athentication tokens:
 
-- **Admin token:** grants full read and write access to all resources in **all organizations in InfluxDB OSS 2.x**.
-  _InfluxDB Cloud does not support Admin tokens._
+- **Operator token:** grants full read and write access to all resources in **all organizations in InfluxDB OSS 2.x**.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/glossary.md:1058]: Curious if we called these "operators" in the UI/cli/api? For full access to everything across orgs, operator doesn't seem as intuitive as admin

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/cloud/reference/api/influxdb-1x/_index.md [hunk 16]
```diff
@@ -14,4 +14,42 @@ related:
   - /influxdb/cloud/query-data/influxql
 ---
 
-{{< duplicate-oss >}}
\ No newline at end of file
+The InfluxDB v2 API includes InfluxDB 1.x compatibility endpoints that work with
```
Comment (russorat) [inline content/influxdb/cloud/reference/api/influxdb-1x/_index.md:17]: can we just explicitly say `/query` and `/write` in this page? Those are the only "compatibility endpoints" i believe.

File: content/influxdb/cloud/reference/api/influxdb-1x/_index.md [hunk 17]
```diff
@@ -14,4 +14,42 @@ related:
...
+InfluxDB 1.x client libraries and third-party integrations like [Grafana](https://grafana.com) and others.
+
+<a class="btn" href="/influxdb/cloud/api/v1-compatibility/">View full v1 compatibility API documentation</a>
+
+## Authentication
+InfluxDB Cloud requires all query and write requests to be authenticated using
```
Comment (russorat) [inline content/influxdb/cloud/reference/api/influxdb-1x/_index.md:23]: what does this have to do with the compatibility endpoints? can we just say "In InfluxDB Cloud, when using the /query or /write apis, you will use your cloud username  as the user and a generated token as the password?

File: content/influxdb/cloud/reference/api/influxdb-1x/_index.md [hunk 18]
```diff
@@ -14,4 +14,42 @@ related:
...
+
+### Token Authentication
+Token authentication requires the following credential:
+
+- **token**: InfluxDB [authentication token](/influxdb/cloud/security/tokens/)
+
+Use the `Authorization` header with the `Token` scheme to provide your
+authentication token to InfluxDB.
+
+##### Token authentication with authorization header
```
Comment (russorat) [inline content/influxdb/cloud/reference/api/influxdb-1x/_index.md:35]: this isn't true for the compatibility apis. you use the token as the password.

File: content/influxdb/cloud/reference/api/influxdb-1x/_index.md (hunk 17 above)
Comment (russorat) [reply content/influxdb/cloud/reference/api/influxdb-1x/_index.md:23]: https://docs.influxdata.com/influxdb/cloud/reference/api/influxdb-1x/#1x-compatible-authorizations

File: content/influxdb/cloud/security/tokens/_index.md [hunk 19]
```diff
@@ -26,6 +26,6 @@ Learn how to create, view, update, or delete an authentication token.
 Grants full read and write access to all resources in an organization.
 
 #### Read/Write token
-Grants read or write access to specific resources in an organization.
+Grants read access, write access, or both to specific resources in an organization.
```
Comment (russorat) [inline content/influxdb/cloud/security/tokens/_index.md:29]: this only works for buckets, not other resources.

File: content/influxdb/v2.0/security/tokens/_index.md [hunk 20]
```diff
@@ -12,7 +12,7 @@ menu:
 weight: 103
 ---
 
-InfluxDB ensures secure interaction between users and data through the use of **authentication tokens**.
+InfluxDB **authentication tokens** ensure secure interaction between users and data.
```
Comment (russorat) [inline content/influxdb/v2.0/security/tokens/_index.md:15]: can we start calling these API tokens?

Comment (sanderson) [reply content/influxdb/v2.0/security/tokens/_index.md:15]: This is a widespread change, so I'm going to reserve this for another PR.

File: content/influxdb/cloud/reference/api/influxdb-1x/_index.md [hunk 21]
```diff
@@ -41,7 +45,18 @@ Authorization: Token <token>
 Authorization: Token mYSuP3rs3cREtT0k3N
 ```
 
-##### InfluxQL support
+### Basic authentication
+Basic authentication requires the following credentials:
+
+- **username**: arbitrary username
```
Comment (russorat) [inline content/influxdb/cloud/reference/api/influxdb-1x/_index.md:51]: even though we technically don't care, maybe we should just ask them to use their cloud2 username here?

**Review by russorat — APPROVED** [review]
LGTM
